docs/main/06-kratix-cli/00_intro.md

```diff
@@ -0,0 +1,212 @@
+
+Use the `--image` flag to add a dependency as a Workflow:
+
+```shell-session
+kratix update dependencies dependencies/ --image myorg/mysql-dependencies:v1.0.0
+```
+
+:::
+
+#### Adding Workflows
+
+To add a Workflow to the Promise, run the [kratix add container](./reference/kratix-add-container) command:
+
+```shell-session
+kratix add container resource/configure/mypipeline --image myorg/mysql-pipeline:v1.0.0
+```
+
+The command above will create a `workflows` directory with some basic files for your
+Pipeline stage. You can then edit the files to add your custom logic.
+
+### From Helm
+
+You can build promises straight from Helm charts. To do so, run the [kratix init
+helm-promise](./reference/kratix-init-helm-promise) command, providing the Chart
+URL. The CLI will auto-generate the Promise API based on the Helm chart values.
+
+For example, consider the [bitnami/mysql Helm
+chart](https://github.com/bitnami/charts/tree/main/bitnami/mysql). To create a
+Promise from this chart, run:
+
+```shell-session
+kratix init helm-promise mysql \
+  --group mygroup.org \
+  --kind Database \
+  --version v1alpha1 \
+  --chart-url oci://registry-1.docker.io/bitnamicharts/mysql
+```
+
+The above command should create a `promise.yaml` file in the current directory that's
+ready to be installed. You can further customise it by adding dependencies and workflows.
+See the [Writing a Promise from Scratch](#from-scratch) section for more details.
+
+### From Operator
+
+You can also build promises from existing Operators. To do so, run the [kratix
+init operator-promise](./reference/kratix-init-operator-promise) command.
+
+For example, consider the [Percona Operator for
+MySQL](https://docs.percona.com/percona-operator-for-mysql/pxc/index.html). To transform
+it into a Promise, first download the Operator manifest bundle to your local machine:
+
+```shell-session
+curl -L https://raw.githubusercontent.com/percona/percona-xtradb-cluster-operator/v1.14.0/deploy/bundle.yaml -o operator-bundle.yaml
+```
+
+Oftentimes, Operators include multiple CRDs. You will need to specify which CRD you want
+to build a Promise from. The Percona Operator comes with a `perconaxtradbclusters.pxc.percona.com` CRD,
+which is used to deploy a MySQL cluster.
+
+:::tip
+
+To find the CustomResourceDefinition (CRD) included in the Operator manifest
+bundle, you can run:
+
+```shell-session
+cat operator-bundle.yaml| grep "CustomResourceDefinition" -A 5 | grep name
+```
+
+:::
+
+Then, run the following command:
+
+```shell-session
+kratix init operator-promise mysql \
+  --group mygroup.org \
+  --kind Database \
+  --version v1alpha1 \
+  --operator-manifests operator-bundle.yaml \
+  --api-schema-from perconaxtradbclusters.pxc.percona.com
+```
+
+:::tip
+
+The `--operator-manifests` flag can be a single file or a directory containing the Operator manifests.
+
+:::
+
+By default, the CLI will create a `promise.yaml` file and embed the Operator bundle in
```

Comment:
Should we mention the error message `The Promise "****" is invalid: metadata.annotations: Too long: must have at most 262144 bytes` because it might not be obvious to people what it is?